client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -435,6 +436,29 @@ public void testDeleteIndex() {
         assertNull(request.getEntity());
     }
 
+    public void testGetSettings() throws IOException {
+        String[] indices = randomIndicesNames(1, 5);
+        GetSettingsRequest getSettingsRequest = new GetSettingsRequest().indices(indices);
+
+        Map<String, String> expectedParams = new HashMap<>();
+        /* HELP
+        setRandomMasterTimeout(getSettingsRequest, expectedParams);
+        I definitely have some questions here.  Enabling this causes the master timeout
+        parameter *not* to come through on the actual request, but I can't figure out
+        why.  Further, I'm not entirely sure I need it.
```

Comment:
that's a good one. the request class supports master timeout but the REST layer never parses it. That is why it doesn't come through. If you'd want it to come through you would need to first pass it in in `Request#testGetSettings` but if you do so you would get back an error saying that the param is not supported by this API. You can have a look at `RestGetSettingsAction` to see what parameters we parse.

I checked if passing in the param using the transport client would do something but it doesn't as this is just a read operation.